### PR TITLE
Basic authentication features

### PR DESCRIPTION
**Pull request by dalisc**
Changes:
- Add authentication api: very simple authentication with just email
- Add Log Out and Log In on navbar
- I have only used the navbar in index.html for now as other pages will change
- @all as you make the other pages, just follow my example in index.html to include the navbar in the page
- Change navbar structure so that it can support authentication globally

TODO: Discuss on Evelyn on user data structure and how that integrates with API

PS: @jacoblipech personal portfolio repo helped me SO MUCH! THANKS HAHA

UPDATE: I realised for the way I did it, the navbar was loading weirdly so I changed the method. Take note that the old navbar import is back. I just inject the authentication portion after the document loads.

### CODE REVIEW

File: src/main/webapp/assets/js/navbar.js [hunk 1]
```diff
@@ -1,12 +1,58 @@
...
+});
+
+/**
+ * Fetch data from /login endpoint to check the status of the user.
+ * Data that the enpoint contains is JSON.
+ * If the user is logged in, the userEmail will be a non-empty string.
+ * Depending on the log in status, the navbar rendered will look different.
+ */
+function isLoggedin() {
+    console.log("trigger")
```
Comment (jirehcwe) [inline src/main/webapp/assets/js/navbar.js:12]: Remove the debugging logs if this function is tested and working!

File: src/main/webapp/assets/js/navbar.js [hunk 2]
```diff
@@ -1,12 +1,58 @@
...
+    fetch('/login').then(response => response.json()).then(userJson => {
+        console.log(userJson);
```
Comment (jirehcwe) [inline src/main/webapp/assets/js/navbar.js:14]: Remove this debug log as well.

File: src/main/webapp/assets/js/navbar.js [hunk 3]
```diff
@@ -1,12 +1,58 @@
...
+
+/**
+ * Fetch data from /login endpoint to check the status of the user.
+ * Data that the enpoint contains is JSON.
+ * If the user is logged in, the userEmail will be a non-empty string.
+ * Depending on the log in status, the navbar rendered will look different.
+ */
+function isLoggedin() {
+    console.log("trigger")
+    fetch('/login').then(response => response.json()).then(userJson => {
```
Comment (jirehcwe) [inline src/main/webapp/assets/js/navbar.js:13]: We decided to stick to async await rather than .then() for Promises!

Comment (dalisc) [reply src/main/webapp/assets/js/navbar.js:13]: my bad! i forgot!

**Pull request merged**